python/tk_multi_publish2/__init__.py

```diff
@@ -27,5 +27,9 @@ def show_dialog(app):
 
     display_name = sgtk.platform.current_bundle().get_setting("display_name")
 
-    # start ui
-    app.engine.show_dialog(display_name, app, AppDialog)
+    if app.pre_publish_hook.pre_publish_validate():
```

Comment:
Maybe add a debug message if validation fails? That would help with debugging when the window doesn't pop up and the hook doesn't have sufficient logging